### PR TITLE
Create Parser header

### PR DESCRIPTION
**Pull request by will889889**
Parser represents a operation that can translate raw message to executable information of a request.

### CODE REVIEW

File: Parser.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+Copyright (c) 2021 Chiu Yen-Chen,
+Swen Sun-Yen.
+All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file. See the AUTHORS file for names of contributors.
+ */
+#pragma once
```
Comment (john0312) [inline Parser.h:8]: Please use #define guards instead. See:
https://google.github.io/styleguide/cppguide.html#The__define_Guard

File: Parser.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#include <string>
+#include "RequestData.h"
+
+class Parser
```
Comment (john0312) [inline Parser.h:12]: Please add class comment. See:
https://google.github.io/styleguide/cppguide.html#Class_Comments

File: Parser.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+{
+public:
+	Parser();
+	~Parser();
+
+	RequestData GetRequestData();
+private:
+	string respData;
```
Comment (john0312) [inline Parser.h:20]: Please use std::string instead of string, because we are not in the std namespace.

File: Parser.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	RequestData reqData;
+	string statusCode;
+	string rawURL;
+	SURL surl;
+
+	void SendResponse();
+	void ParseRequest();
+};
```
Comment (john0312) [inline Parser.h:28]: Newline at the end.

File: Parser.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	Parser();
+	~Parser();
+
+	RequestData GetRequestData();
+private:
+	string respData;
+	RequestData reqData;
+	string statusCode;
+	string rawURL;
+	SURL surl;
```
Comment (john0312) [inline Parser.h:24]: Include the header that defines SURL or forward declares it above.

File: Parser.h [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+class Parser
+{
+public:
+	Parser();
+	~Parser();
+
+	RequestData GetRequestData();
+private:
+	string respData;
+	RequestData reqData;
```
Comment (john0312) [inline Parser.h:21]: Include the header that defines RequestData or forward declares it above.

File: Parser.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+#include <string>
+#include "RequestData.h"
+
+class Parser
+{
+public:
+	Parser();
+	~Parser();
+
+	RequestData GetRequestData();
```
Comment (john0312) [inline Parser.h:18]: Please add comments for the method, see:
https://google.github.io/styleguide/cppguide.html#Function_Comments

**Comment by john0312** [comment]
Hello @destinycan (Zoan), please take a look at this commit/PR. Thank you.

**Comment by will889889** [comment]
Done

**Comment by destinycan** [comment]
LGTM

**Pull request merged**